examples/code_highlight.py

```diff
@@ -0,0 +1,32 @@
+from PyQt5.QtGui import QColor, QPalette
+from qtpy.QtWidgets import QApplication, QTextEdit
+
+from superqt.utils import CodeSyntaxHighlight
+
+app = QApplication([])
+
+text_area = QTextEdit()
+
+highlight = CodeSyntaxHighlight(text_area.document(), "python", "monokai")
+
+palette = text_area.palette()
+palette.setColor(QPalette.Base, QColor(highlight.background_color))
```

Comment:
this seems like it might be something that happens a lot huh?  Do you think we should add a `set_background` parameter to `CodeSyntaxHighlight` that does this automatically?  (default could be either True or False)

Reply:
You mean to pass `QTextEdit` instead of `QTextDocument` to `CodeSyntaxHighlight` constructor? 
Or allow both types and use `set_background` only if `QTextEdit` is passed?

Reply:
Or maybe add method `setTextEditBackground(self, test_edit: QTextEdit)` ?

Reply:
ohhh, never mind me :)  I missed the fact that it's `text_area.document()`
let's leave that bit of magic for later.  The example is good enough for now